fetch.bs

```diff
@@ -3035,17 +3051,31 @@ steps:
 <i>authentication-fetch flag</i>.
 
 <ol>
+ <li><p>Let <var>httpRequest</var> be null.
+
+ <li><p>If <var>request</var>'s <a for=request>window</a> is "<code>no-window</code>" and
+ <var>request</var>'s <a for=request>redirect mode</a> is "<code>error</code>", then set
+ <var>httpRequest</var> to <var>request</var>.
+
  <li>
-  <p>Let <var>httpRequest</var> be <var>request</var> if
-  <var>request</var>'s <a for=request>window</a> is "<code>no-window</code>"
-  and <var>request</var>'s <a for=request>redirect mode</a> is
-  "<code>error</code>", and the result of <a lt=clone for=request>cloning</a>
-  <var>request</var> otherwise.
+  <p>Otherwise, run these substeps:
 
-  <p class="note no-backref">A <var>request</var> is typically cloned as it needs to be possible to
-  add <a>headers</a> and read its
-  <a for=request>body</a> without affecting <var>request</var>. As
-  <var>request</var> is reused with redirects, authentication, and proxy authentication.
+  <ol>
+   <li><p>Set <var>httpRequest</var> to a copy of <var>request</var> except for its
+   <a for=request>body</a>.
+
+   <li><p>Let <var>body</var> be <var>request</var>'s <a for=request>body</a>.
+
+   <li><p>Set <var>httpRequest</var>'s <a for=request>body</a> to <var>body</var>.
+
+   <li><p>If <var>body</var> is non-null, then set <var>request</var>'s <a for=request>body</a> to a
+   new <a for=/>body</a> whose <a for=body>stream</a> is null and whose <a for=body>source</a> is
+   <var>body</var>'s <a for=body>source</a>.
+  </ol>
+
+  <p class="note no-backref">Here we do not <a for=request>clone</a> <var>request</var> in order
+  to reduce memory consumption. <var>request</var> can be reused with redirects, authentication,
+  and proxy authentication.
```

Comment:
I'd say "request's body's source is null" or "request's body is made from ReadableStream" insetad of "request's body's source is a ReadableStream object".